### PR TITLE
fix: re-use submitted DA batches on SL failure

### LINKED ISSUE #657: Non-atomic batch submission to data availability and settlement layers causes repaying for the same data and potential indefinite failure

### PR DESCRIPTION
**Pull request by mtsitrin**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

Close #657 

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

**Comment by danwt** [comment]
Could you link the closed issue?

**Review by danwt — DISMISSED** [review]
I guess it works but somehow the dymint code is becoming more and more spaghetti to handle all these cases

Probably what we need is a pipeline

Block produced -> Submit to DA -> Submit to settlement
With queues (channels) connecting

File: block/submit.go [hunk 1]
```diff
@@ -61,50 +61,65 @@ func (m *Manager) handleSubmissionTrigger(ctx context.Context) {
 }
 
 func (m *Manager) submitNextBatch() (uint64, error) {
-	// Get the batch start and end height
-	startHeight := m.syncTarget.Load() + 1
-	endHeight := uint64(m.store.Height())
+	//check if pending batch exists.
+	if m.pendingBatch != nil {
```
Comment (omritoptix) [inline block/submit.go:65]: Possibly be cleaner to extract all the batch fetching/creation to an external method as this method currently does 3 things:
1. creates batch (or gets pending one)
2. submits to DA
3. submit to Settlement.

**Review by omritoptix — COMMENTED** [review]
Is this path tested in the test cases? i.e of `pendingBatch` != nil ?

File: block/submit.go [hunk 2]
```diff
@@ -49,37 +49,64 @@ func (m *Manager) handleSubmissionTrigger(ctx context.Context) {
...
+			return
+		}
+
+		resultSubmitToDA, err := m.submitNextBatchToDA(nextBatch)
+		if err != nil {
+			m.logger.Error("submit next batch", "error", err)
+			return
+		}
+
+		m.pendingBatch = &struct {
```
Comment (omritoptix) [inline block/submit.go:67]: I think would be cleaner to define the struct not in-place as it requires re-declearing the fields. or alternatively initiate it on the newManager vs on every submit attempt

**Review by omritoptix — APPROVED** [review]

**Pull request merged**